### DISCUSSION TITLE
Missing property oauth2 in google.accounts in [@types/gsi]

### DISCUSSION THREAD

**Discussion by RichardJECooke** [DefinitelyTyped/DefinitelyTyped#61379] [Issues with a @types package]
Your types definition on `google.accounts` is missing property `oauth2`.

This sample code `const tokenClient = google.accounts.oauth2.initTokenClient` straight from [Google Drive Javascript Quickstart](https://developers.google.com/drive/api/quickstart/js) doesn't work with your definitions.

Version `"@types/gsi": "^0.0.0"`.

[a comment by an automated account is not shown]

**Comment by aronhoyer** [comment]
that's interesting. the `oauth2` prop is missing from the docs of the [gsi javascript client reference](https://developers.google.com/identity/gsi/web/reference/js-reference) as well. i've submitted feedback on the gsi client reference. personally i'd wait till the docs are updated, so that we have the full definition of the object and the methods within it, before the type definitions are updated.

in the meantime @RichardJECooke, you could always extend the definition in your own `.d.ts` and add the props you need.

**Reply by j-frost** [reply]
There are additional docs that detail `google.accounts.oauth2` at [developers.google.com/identity/oauth2/web/reference/js-reference](https://developers.google.com/identity/oauth2/web/reference/js-reference). Is that enough to add it to the typing? 

edit: In a roundabout way, this does make sense. GSI is intended to separate authentication and authorization; the section you're referencing is filed under the authentication section of the docs. Does make it somewhat obtuse though what the lib can actually do...

**Comment by filmerjarred** [comment]
My quick work-around:

```
interface initCodeClient {
  client_id: string;
  scope: string;
  ux_mode: "popup" | "redirect";
  callback: ({ code: string, scope: string }) => void;
}

declare global {
  interface Window {
    google: {
      accounts: {
        oauth2: {
          initCodeClient: (arg0: initCodeClient) => any;
        };
      };
    };
  }
}
```
